lib/index.js

```diff
@@ -37,6 +37,11 @@ function getCookieOptions({ userCookieOptions, ttl }) {
   };
 }
 
+/**
```

Comment:
I am new to TypeScript (not using it), is it necessary to add jsDoc to the source file too? Can't we only have the index.d.ts and manually update it when needed?

Reply:
The declaration generation is based on the definitions, as there's only so much it can be smart about with javascript. like the cookieName it deduced to be void becuase of the throwIfNoName implementation. 

You can definitely keep the declaration in in sync manually, save from all the extra deps, may be the most appropriate for this small lib. 

Let me improve on it and raise another pr in the upcoming days

Reply:
Awesome! Yes ideally I'd like to only have the declaration file and being able to manually sync it/check if it's in sync.

@neoziro, any advice on adding declaration files to pure JavaScript libraries? I don't want to move this library to TypeScript, just want the experience for TypeScript users to be good (whatever that means, I have a good experience already with pure JS :D)

Reply:
I don’t know I never tried.

Typing a library correctly is a challenge. If your library is not written in TypeScript and if you don’t have any TypeScript experience, then types would be very weak and probably not good.

Example: using a string instead of an enum, no generics, etc..